corehq/apps/reports/standard/deployments.py

```diff
@@ -209,8 +210,33 @@ def user_query(self, pagination=True):
                                        )
         return user_query
 
+    def get_location_columns(self, grouped_ancestor_locs):
+        from corehq.apps.locations.models import LocationType
+        location_types = LocationType.objects.by_domain('icds-cas')
+        all_user_locations = grouped_ancestor_locs.values()
+        all_user_loc_types = {loc.location_type_id for user_locs in all_user_locations for loc in user_locs}
+        required_loc_columns = [loc_type for loc_type in location_types if loc_type.id in all_user_loc_types]
+        return required_loc_columns
+
+    def user_locations(self, ancestors, location_type):
+        location_list = ['---'] * len(location_type)
+        types_id = [type.id for type in location_type]
+        for loc in ancestors:
+            position = types_id.index(loc.location_type_id)
+            location_list[position] = loc.name
+        return location_list
```

Comment:
You could also avoid using indices by writing this method as
```python
def user_locations(self, ancestors, location_types):
    ancestors_by_type_id = {loc.location_type_id: loc.name for loc in ancestors}
    return [
        ancestors_by_type_id.get(location_type.id, '---')
        for location_type in location_types
    ]
```

Reply:
awesome thank!. Thats certainly smarter implementation.